extensions/amp-story/1.0/test/test-media-performance-metrics-service.js

```diff
@@ -26,11 +26,7 @@ describes.fakeWin('media-performance-metrics-service', {amp: true}, (env) => {
   let win;
 
   before(() => {
-    clock = lolex.install({
-      target: win,
-      toFake: ['Date'],
-      now: 0,
-    });
+    clock = fakeTimers.withGlobal(win).install({toFake: ['Date'], now: 0});
```

Comment:
@gmajoulet @Enriqe This change is supposed to be straightforward, but is [causing this test to fail](https://travis-ci.com/github/ampproject/amphtml/jobs/473555389#L624). It seems like this is the only test file that initializes `clock` in `before()` instead of `beforeEach()`. Changing the initialization site causes other test failures. Can one of you help figure out why these tests are failing with the new timer class?

```sh
DESCRIBE => media-performance-metrics-service
  DESCRIBE =>  
    IT => "before all" hook for "should record and flush metrics"
      ✗ TypeError: Cannot read property 'navigator' of undefined
          at Object.withGlobal (/tmp/node_modules/@sinonjs/fake-timers/src/fake-timers-src.js:7:29)
          at Context.<anonymous> (/tmp/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js:29:24)
Chrome Headless 88.0.4324.96 (Linux x86_64) media-performance-metrics-service   "before all" hook for "should record and flush metrics" FAILED
	TypeError: Cannot read property 'navigator' of undefined
	    at Object.withGlobal (/tmp/node_modules/@sinonjs/fake-timers/src/fake-timers-src.js:7:29)
	    at Context.<anonymous> (/tmp/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js:29:24)
●
    IT => "after all" hook for "should not flush metrics if sendMetrics is false"
      ✗ TypeError: Cannot read property 'uninstall' of undefined
          at Context.<anonymous> (/tmp/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js:33:11)
Chrome Headless 88.0.4324.96 (Linux x86_64) media-performance-metrics-service   "after all" hook for "should not flush metrics if sendMetrics is false" FAILED
	TypeError: Cannot read property 'uninstall' of undefined
	    at Context.<anonymous> (/tmp/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js:33:11)
```

Reply:
Unfortunately I don't know anything about this new timer class and the API changes. If you're stuck I can help you look into it but to be honest I'll need a few days...

Reply:
No hurry. Since the new timer class contains some fixes that Dima was asking for, I'll temporarily skip these amp-story tests in order to unblock progress.

Reply:
These tests should not be skipped, a lot of internal infrastructure and cross-PAs architecture decision are being taken using the metrics they make sure are correct. I'll do my best to look into it tomorrow.

Reply:
SGTM. Here are the failures we're looking to fix:
- [Travis](https://travis-ci.com/github/ampproject/amphtml/jobs/473992562#L583)
- [CircleCI](https://app.circleci.com/pipelines/github/ampproject/amphtml/206/workflows/ebf9b80c-28ad-4664-91a8-eabaca533db1/jobs/1308)

And here is the diff that is causing the tests to break:
```diff
diff --git a/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js b/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js
index 3c3f7c3d6..b95848d4c 100644
--- a/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js
+++ b/extensions/amp-story/1.0/test/test-media-performance-metrics-service.js
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import * as lolex from 'lolex';
+import * as fakeTimers from '@sinonjs/fake-timers';
 import {MEDIA_LOAD_FAILURE_SRC_PROPERTY} from '../../../../src/event-helper';
 import {MediaPerformanceMetricsService} from '../media-performance-metrics-service';
 import {Services} from '../../../../src/services';
@@ -26,11 +26,7 @@ describes.fakeWin('media-performance-metrics-service', {amp: true}, (env) => {
   let win;
 
   before(() => {
-    clock = lolex.install({
-      target: win,
-      toFake: ['Date'],
-      now: 0,
-    });
+    clock = fakeTimers.withGlobal(win).install({toFake: ['Date'], now: 0});
   });
 
   after(() => {
```

Reply:
I believe I've found the bug. As originally written, `before()` was using an uninitialized `win` object to initialize `clock`. This somehow worked with `lolex` v5, but `@sinonjs/fake-timers` v7 [checks](https://github.com/sinonjs/fake-timers/blob/v7.0.2/src/fake-timers-src.js#L7) to make sure the fields are non-null. I've fixed the bug by initializing `clock` using `window`, which is available in `before()`. Will merge this once all CI checks are green.